incremental.go

```diff
@@ -0,0 +1,200 @@
+package feedx
+
+import (
+	"context"
+	"time"
+
+	"github.com/bsm/bfs"
+)
+
+// IncrmentalProduceFunc returns a ProduceFunc closure around an incremental mod time.
+type IncrementalProduceFunc func(time.Time) ProduceFunc
+
+// IncrementalProducer produces a continuous incremental feed.
+type IncrementalProducer struct {
+	*ProducerState
```

Comment:
you can just embed it as `producerState` as long as the methods are pointer methods, it will do the right thing